ambry-router/src/main/java/com.github.ambry.router/GetBlobOperation.java

```diff
@@ -877,6 +939,61 @@ MessageFormatFlags getOperationFlag() {
           : MessageFormatFlags.All;
     }
 
+    @Override
+    protected boolean maybeProcessDecryptCallbackAndCompleteOperation() {
+      if (decryptionRequired.get() && decryptCallbackResultInfo.resultAvailable) {
+        // in case of Metadata blob, only user-metadata needs decryption if the blob is encrypted
```

Comment:
Does this method really have to be overridden?

Reply:
if we resort to decrypt both content and userMetadata, we don't need to.